### PR TITLE
feat(sast): create a go package and a runner, add a test to create to so

### PR DESCRIPTION
**Pull request by omryMen**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sast/go_runner.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    # send the document as a byte array of json format
+    entire_file_output = analyze_code(json.dumps(document).encode('utf-8'))
+
+    # we dereference the pointer to a byte array
+    entire_file_bytes = ctypes.string_at(entire_file_output)
+
+    # convert our byte array to a string
+    entire_file_string = entire_file_bytes.decode('utf-8')
+    print(entire_file_output, entire_file_bytes, entire_file_string)
+    return json.loads(entire_file_string)
```
Comment (achiar99) [inline checkov/sast/go_runner.py:29]:
```suggestion
    analyze_code_output = analyze_code(json.dumps(document).encode('utf-8'))

    # we dereference the pointer to a byte array
    analyze_code_bytes = ctypes.string_at(analyze_code_output)

    # convert our byte array to a string
    analyze_code_string = analyze_code_bytes.decode('utf-8')
    print(analyze_code_output, analyze_code_bytes, analyze_code_string)
    return json.loads(analyze_code_string)
```

**Review by achiar99 — COMMENTED** [review]
Nice
Lets create in the checkov side `semgrep_runner` and we can decide in the run function which one to run
Also lets create the so file manually for now

**Comment by achiar99** [comment]
Also maybe add https://github.com/smacker/go-tree-sitter to the sum file and create also the mod file

File: checkov/sast/go_runner.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    policy_file: str,
+                    policy_dir: str,
+                    language: str):
+    if not source_code_file and not source_code_dir:
+        raise Exception('must provide source code file or dir for sast runner')
+    if not policy_dir and not policy_file:
+        raise Exception('must provide policy file or dir for sast runner')
+    if not language:
+        raise Exception('must provide a language for sast runner')
+    if language not in AVAILABLE_LANGUAGES:
```
Comment (achiar99) [inline checkov/sast/go_runner.py:49]: we already check that in the `runner_filter`

File: checkov/sast/go_runner.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+analyze_code = library.analyzeCode
+analyze_code.restype = ctypes.c_void_p
+
+
+def run_go_library(source_code_file: str,
+                   source_code_dir: str,
+                   policy_file: str,
+                   policy_dir: str,
+                   language: str) -> Dict[str, Any]:
+    validate_params(**locals())
```
Comment (achiar99) [inline checkov/sast/go_runner.py:19]:
```suggestion
def run_go_library(language: str
                   source_code_file: Optional[str] = None,
                   source_code_dir: Optional[str] = None,
                   policy_file: Optional[str] = None,
                   policy_dir: Optional[str] = None) -> Dict[str, Any]:
    validate_params(**locals())
```

**Review by achiar99 — APPROVED** [review]
💯

**Review by arielkru — APPROVED** [review]
LGTM

**Pull request merged**